lens-cube/src/main/java/org/apache/lens/cube/parse/StorageCandidate.java

```diff
@@ -137,9 +181,340 @@ public boolean contains(Candidate candidate) {
     return null;
   }
 
+  private void updatePartitionStorage(FactPartition part) throws LensException {
```

Comment:
We don't need backward compatibility  here. This is just internal code and not a feature that is exposed to end user. Please fix and make sure all usages are taken into account.